numpy/testing/_private/utils.py

```diff
@@ -1699,10 +1704,10 @@ def assert_array_almost_equal_nulp(x, y, nulp=1):
     ref = nulp * np.spacing(np.where(ax > ay, ax, ay))
     if not np.all(np.abs(x-y) <= ref):
         if np.iscomplexobj(x) or np.iscomplexobj(y):
-            msg = "X and Y are not equal to %d ULP" % nulp
+            msg = f"Arrays are not equal to {nulp} ULP"
         else:
             max_nulp = np.max(nulp_diff(x, y))
-            msg = "X and Y are not equal to %d ULP (max is %g)" % (nulp, max_nulp)
+            msg = f"Arrays are not equal to {nulp} ULP (max is {max_nulp:.0f})"
```

Comment:
Switched to f-string to fix PEP8 line length

Reply:
LGTM, but I am not sure that `:.0f` is much better.  I suspect `max_nulp` can be a huge number when they don't match.  OTOH, we do expect a small number, so also OK.

Reply:
You're right though; `g` is better because it automatically switches to scientific notification in those cases.
```suggestion
            msg = f"Arrays are not equal to {nulp} ULP (max is {max_nulp:g})"
```